db/migrate/20180606155924_move_ansible_container_secrets_into_database.rb

```diff
@@ -14,14 +14,26 @@ def up
     update_authentications
   end
 
-  def self.read_token(file)
-    File.read(file)
+  private
+
+  def token_file
+    if Rails.env.test? && ENV['TOKEN_FILE']
+      ENV['TOKEN_FILE']
+    else
+      TOKEN_FILE
+    end
   end
 
-  private
+  def ca_cert_file
+    if Rails.env.test? && ENV['CA_CERT_FILE']
+      ENV['CA_CERT_FILE']
+    else
+      CA_CERT_FILE
```

Comment:
These methods were added to inject different behavior in test vs production.  See b5cb2e7a857e2886636f3fd0142cf642b2084e5d

Reply:
can we change the definition of CA_CERT_FILE instead?

```
CA_CERT_FILE = ENV[] || "const".freeze
```

Reply:
no. ignore

can we just do:

```ruby
def ca_cert_file
  ENV.fetch("CA_CERT_FILE", "/run/secrets/kubernetes.io/serviceaccount/ca.crt")
end
```

or maybe `ENV[].presence || ""`

Reply:
> and do these need to be test only?

Yes, my objective was to not change the migration to add test code into it.  I think what is here is a fair enough workaround to avoid test code in migrations while still being able to work with rails 7 neutering the ability to do `allow`/`expect` on the stub migration models.

Reply:
> ```ruby
> def ca_cert_file
>   ENV.fetch("CA_CERT_FILE", "/run/secrets/kubernetes.io/serviceaccount/ca.crt")
> end
> ```

I was trying not to change the behavior of production migrations.  While the proposed code reads, I really didn't want to change the behavior of past migrations.